frontend/src/app/data-entry/location-form/location-form.component.ts

```diff
@@ -1,55 +1,128 @@
                 {
-                    link: `/data-entry/sources/${data.spaceDescription.source.id}`,
-                    label: data.spaceDescription.source.name
+                    link: `/data-entry/sources/${location.source.id}`,
+                    label: location.source.name,
                 },
                 {
-                    link: `/data-entry/locations/${data.spaceDescription.id}`,
-                    label: data.spaceDescription.name
+                    link: `/location-entry/locations/${location.id}`,
+                    label: location.name,
                 },
-            ];
-        } else {
-            return [
-                { link: '/', label: 'Lettercraft' },
-                { link: '/data-entry', label: 'Data entry' },
-                { link: '', label: 'Location not found' }
             ]
-        }
+        })
+    )
+
+
+    public dataIcons = dataIcons;
+    public deletingInProgress = false;
+
+    constructor(
+        private destroyRef: DestroyRef,
+        private router: Router,
+        private modalService: ModalService,
+        private toastService: ToastService,
+        private formService: FormService,
+        private locationQuery: DataEntrySpaceDescriptionGQL,
+        private deleteLocation: DataEntryDeleteLocationGQL
+    ) { }
+
+    public onClickDelete(location: QueriedLocation): void {
+        this.modalService
+            .openConfirmationModal({
+                title: "Delete location",
+                message: `Are you sure you want to delete this location? (${location.name})`,
+            })
+            .then(() => this.performDelete(location.id, location.source.id))
+            .catch(() => {
+                // Do nothing on cancel / dismissal.
+            });
+    }
+
+    private performDelete(locationId: string, sourceId: string): void {
+        this.deletingInProgress = true;
+        this.deleteLocation
+            .mutate(
+                {
+                    id: locationId,
+                },
+                {
+                    update: (cache) => this.updateCache(cache, locationId),
+                }
+            )
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((result) => {
+                this.deletingInProgress = false;
+                const errors = result.data?.deleteSpace?.errors;
+                if (errors && errors.length > 0) {
+                    this.toastService.show({
+                        body: errors.map((error) => error.messages).join("\n"),
+                        type: "danger",
+                        header: "Deletion failed",
+                    });
+                } else {
+                    this.toastService.show({
+                        body: "Location deleted",
+                        type: "success",
+                        header: "Success",
+                    });
+                }
+                this.router.navigate([`/data-entry/sources/${sourceId}`]);
```

Comment:
For long nested functions like this, I find it more readable if they're moved into their own function/method.

Reply:
I agree; again this was copied from a branch that did not yet have all the outfactors to separate functions.